api/types/access_request.go

```diff
@@ -776,6 +781,28 @@ func (f *AccessRequestFilter) FromMap(m map[string]string) error {
 
 // Match checks if a given access request matches this filter.
 func (f *AccessRequestFilter) Match(req AccessRequest) bool {
+	// only return if the request was made by the api requester
+	if f.Scope == AccessRequestScope_MY_REQUESTS && req.GetUser() != f.Requester {
+		return false
+	}
+	// a user cannot review their own requests
+	if f.Scope == AccessRequestScope_NEEDS_REVIEW && req.GetUser() == f.Requester {
+		return false
+	}
```

Comment:
is this missing a condition? `needs_review` means any request that is `not my own` and where `state` is not `approved or denied or promoted`? (or state = pending)

Reply:
Yeah i think that makes sense. I'll add the pending check